sympy/core/sympify.py

```diff
@@ -88,8 +90,8 @@ def _convert_numpy_types(a, **sympify_args):
         prec = np.finfo(a).nmant + 1
```

Comment:
Are we sure this logic is correct?

```
>>> a = np.float32(232132.2)
>>> a
232132.2
>>> sympify(a)
232132.
>>> Float('232132.2', precision=24)
232132.
>>> Float('232132.2', precision=25)
232132.2
```

Not completely sure why this happens. Some discrepancy between how a mpmath and machine float are implemented.

Reply:
Or maybe this is an issue with sympy

```
>>> Float('232132.2', precision=24)._mpf_
(0, mpz(14856461), -6, 24)
>>> Float('232132.2', precision=25)._mpf_
(0, mpz(14856461), -6, 24)
```

Reply:
I don't think that numpy's nmant is consistent: https://github.com/numpy/numpy/issues/22333



> Or maybe this is an issue with sympy
> 
> ```
> >>> Float('232132.2', precision=24)._mpf_
> (0, mpz(14856461), -6, 24)
> >>> Float('232132.2', precision=25)._mpf_
> (0, mpz(14856461), -6, 24)
> ```

The `_mpf_` is not expected to be affects by the precision argument.

> Are we sure this logic is correct?
> 
> ```
> >>> a = np.float32(232132.2)
> >>> a
> 232132.2
> >>> sympify(a)
> 232132.
> >>> Float('232132.2', precision=24)
> 232132.
> >>> Float('232132.2', precision=25)
> 232132.2
> ```

This is to do with printing:
```
In [8]: a = np.float32(232132.2)

In [9]: sa = sympify(a)

In [10]: sa
Out[10]: 232132.

In [11]: sa._prec
Out[11]: 24

In [12]: Float(sa, precision=25)
Out[12]: 232132.2
```